mongo/mongo.go

```diff
@@ -517,14 +517,15 @@ func ensureServer(args EnsureServerParams, mongoKernelTweaks map[string]string)
 	if installed {
 		exists, err := svc.Exists()
 		if err != nil {
-			return errors.Trace(err)
+			errors.Trace(err)
```

Comment:
This needs to be return errors.Trace(err), otherwise the error just gets dropped on the floor.

Oh, I think I see - you're deliberately ignoring the error here? I had expected you to change the implementation of `systemd.Service.Exists` so that it would return `(false, nil)` when there's no service conf file in `/lib/systemd/system` but there is one under `/var/lib/juju/init`. (That might require passing the fallback directory into `systemd.NewService`.)

Ignoring the error here is too broad - it means that you might ignore different errors that we don't expect, leading to hard-to-find bugs, and it also ignores errors from the upstart init system, when this should be systemd specific.